src/workerd/io/io-own.c++

```diff
@@ -33,6 +33,38 @@ void DeleteQueue::scheduleDeletion(OwnedObject* object) const {
   }
 }
 
+void DeleteQueue::scheduleAction(jsg::Lock& js, kj::Function<void(jsg::Lock&)>&& action) const {
+  if (IoContext::hasCurrent() && IoContext::current().deleteQueue.get() == this) {
+    // Same thread! No need to enqueue.
+    action(js);
```

Comment:
IMO even if we're already in the right context, it would be better to enqueue the action for asynchronous execution the same way as if we weren't. Otherwise, in some cases it can be tricky to write the callback in a way that's safe to run both synchronously and asynchronously. The code below that schedules cross-context should work just fine even if we happen to be in the same context.

Our target use case won't ever schedule on the same context anyway, of course, so it can't hurt there.